source/frontend/src/models/generate/lease/GenerateLease.test.ts

```diff
@@ -0,0 +1,277 @@
+import { ContactMethodTypes } from '@/constants/contactMethodType';
+import { Api_Insurance } from '@/models/api/Insurance';
+import { Api_Lease } from '@/models/api/Lease';
+import { Api_LeaseTenant } from '@/models/api/LeaseTenant';
+import { Api_Property } from '@/models/api/Property';
+
+import { Api_GenerateLease } from './GenerateLease';
+
+describe('GenerateLease tests', () => {
+  it('generates an empty lease without throwing an error', () => {
+    const lease = new Api_GenerateLease({} as Api_Lease, [], [], []);
+    expect(lease.file_number).toBe('');
+  });
+
+  it('generates a lease with a commencement_date', () => {
+    const lease = new Api_GenerateLease(
+      { terms: [{ startDate: '01-01-2020' }, { startDate: '02-02-2022' }] } as Api_Lease,
+      [],
+      [],
+      [],
+    );
+    expect(lease.commencement_date).toBe(`January 01, 2020`);
+  });
+
+  it('generates a lease with a full land_string', () => {
+    const lease = new Api_GenerateLease(
+      {
+        properties: [
+          {
+            property: { pid: 1, landLegalDescription: 'test' } as Api_Property,
+            leaseArea: 1,
+            areaUnitType: { description: 'square feet' },
+          },
+        ],
+      } as Api_Lease,
+      [],
+      [],
+      [],
+    );
+    expect(lease.land_string).toBe(`Parcel Identifier: 000-000-001\ntest\n1 square feet`);
+  });
+
+  it('generates a lease with a default area unit', () => {
+    const lease = new Api_GenerateLease(
+      {
+        properties: [
+          {
+            property: { pid: 1, landLegalDescription: 'test' } as Api_Property,
+            leaseArea: 1,
+            areaUnitType: null,
+          },
+        ],
+      } as Api_Lease,
+      [],
+      [],
+      [],
+    );
+    expect(lease.land_string).toBe(`Parcel Identifier: 000-000-001\ntest\n1 m²`);
+  });
+  it('generates a lease with no pid', () => {
+    const lease = new Api_GenerateLease(
+      {
+        properties: [
+          {
+            property: { pid: undefined, landLegalDescription: 'test' } as Api_Property,
+            leaseArea: 1,
+            areaUnitType: { description: 'square feet' },
+          },
+        ],
+      } as Api_Lease,
+      [],
+      [],
+      [],
+    );
+    expect(lease.land_string).toBe(`Parcel Identifier: \ntest\n1 square feet`);
+  });
+
+  it('generates a lease with no legal description', () => {
+    const lease = new Api_GenerateLease(
+      {
+        properties: [
+          {
+            property: { pid: undefined, landLegalDescription: 'test' } as Api_Property,
+            leaseArea: 1,
+            areaUnitType: { description: 'square feet' },
+          },
+        ],
+      } as Api_Lease,
+      [],
+      [],
+      [],
+    );
+    expect(lease.land_string).toBe(`Parcel Identifier: \ntest\n1 square feet`);
+  });
```

Comment:
This seems a repeat from the test above - but the name implies you want to test for an empty legal description. please fix